dpctl/_sycl_context.pyx

```diff
@@ -335,6 +336,13 @@ cdef class SyclContext(_SyclContext):
         else:
             return False
 
+    def __hash__(self):
+        """
+        Returns hash value corresponding to ``self._ctxt_ref``.
```

Comment:
Most probably the comment was introduced by me, but how about: "Returns a has value by hashing the underlying SYCL context object." Similar changes elsewhere too.